### PR TITLE
Update configmap class constructor and documentation

### PR DESCRIPTION
**Pull request by oharan2**
Fix class constructor to not demand upper (super) classes args, fixed documentation

### CODE REVIEW

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/configmap.py [hunk 1]
```diff
@@ -1,34 +1,23 @@
...
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            data (dict): key-value configuration pairs.
```
Comment (rnetser) [inline ocp_resources/configmap.py:18]: should raise if missing (this is mandatory right?)

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/configmap.py (hunk 1 above)
Comment (oharan2) [reply ocp_resources/configmap.py:18]: Done

**Review by rnetser — APPROVED** [review]

**Pull request merged**